app/experimenter/nimbus-ui/src/components/SlugTextControl/slugify.ts

```diff
@@ -0,0 +1,15 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+export function slugify(text?: string) {
+  if (!text) return "";
+  return text
+    .toString()
+    .toLowerCase()
+    .replace(/\s+/g, "-") // Replace spaces with -
+    .replace(/[^\w-]+/g, "") // Remove all non-word chars
+    .replace(/--+/g, "-") // Replace multiple - with single -
+    .replace(/^[-_]+/, "") // Trim - from start of text
+    .replace(/[-_]+$/, ""); // Trim - from end of text
+}
```

Comment:
Oh interesting........ I think the serializer currently expects that we don't explicitly pass in the slug and it's auto generated....

Yep: https://github.com/mozilla/experimenter/blob/main/app/experimenter/experiments/api/v5/serializers.py#L576

So... when we go and hook this up we can either let the frontend to the slug and bypass that on the server, or only pass name and let the server do the slugification and just let this be a handy preview but then in theory they might not match...  Yeah I dunno couple possibilities.

Reply:
Oh yeah, this doesn't get submitted at all. This was just in the design as a preview. Tried to keep it close to Django's impl, but either way the server wins. This value doesn't even get passed back in the `onSave` callback to SidebarActions